bindings/pulp/bindings/server.py

```diff
@@ -289,47 +310,68 @@ def request(self, method, url, body):
+                else:
+                    continue
+            if auth_type is 'Certificate':
+                if self.pulp_connection.cert_filename:
+                    ssl_context.load_cert(self.pulp_connection.cert_filename)
+                else:
+                    continue
+            if auth_type is 'OAuth':
+                # oauth configuration. This block is only True if oauth is not None, so it won't run on RHEL
+                # 5.
+                if self.pulp_connection.oauth_key and self.pulp_connection.oauth_secret and oauth:
+                    oauth_consumer = oauth.Consumer(
+                        self.pulp_connection.oauth_key,
+                        self.pulp_connection.oauth_secret)
+                    oauth_request = oauth.Request.from_consumer_and_token(
+                        oauth_consumer,
+                        http_method=method,
+                        http_url='https://%s:%d%s' % (self.pulp_connection.host, self.pulp_connection.port, url))
+                    oauth_request.sign_request(oauth.SignatureMethod_HMAC_SHA1(), oauth_consumer, None)
+                    oauth_header = oauth_request.to_header()
+                    # unicode header values causes m2crypto to do odd things.
+                    for k, v in oauth_header.items():
+                        oauth_header[k] = encode_unicode(v)
+                    headers.update(oauth_header)
+                    headers['pulp-user'] = self.pulp_connection.oauth_user
+                else:
+                    continue
+            if auth_type is 'Basic':
+                if self.pulp_connection.username and self.pulp_connection.password:
+                    raw = ':'.join((self.pulp_connection.username, self.pulp_connection.password))
+                    encoded = base64.encodestring(raw)[:-1]
+                    headers['Authorization'] = 'Basic ' + encoded
+                else:
+                    continue
+
+            connection = httpslib.HTTPSConnection(
+                self.pulp_connection.host, self.pulp_connection.port, ssl_context=ssl_context)
+            try:
+                # Request to the server
+                connection.request(method, url, body=body, headers=headers)
+                response = connection.getresponse()
+            except SSL.SSLError, err:
+                # Translate stale login certificate to an auth exception
+                if 'sslv3 alert certificate expired' == str(err):
+                    raise exceptions.ClientCertificateExpiredException(
+                        self.pulp_connection.cert_filename)
+                elif 'certificate verify failed' in str(err):
+                    raise exceptions.CertificateVerificationException()
+                else:
+                    raise exceptions.ConnectionException(None, str(err), None)
+                
+            if response.status != 401:
+                break
```

Comment:
It seems like we should make sure that at least one of these were successful. If neither were successful, the loop will still exit and the later code will execute next. This may explain why I saw issues when I had no authentication method.